src/main/java/dev/revere/alley/base/server/ServerService.java

```diff
@@ -115,4 +148,73 @@ public void clearAllQueues(Player issuer) {
             }
         }
     }
+
+    @Override
+    public List<Material> getBlockedCraftingItems() {
+        return Collections.unmodifiableList(this.blockedCraftingItems);
+    }
+
+    @Override
+    public void loadBlockedCraftingMaterials() {
+        FileConfiguration config = this.configService.getSettingsConfig();
+        List<String> blocked = config.getStringList(this.BLOCKED_MATERIALS_SECTION);
+
+        this.blockedCraftingItems.clear();
+        for (String mat : blocked) {
+            try {
+                Material material = Material.valueOf(mat);
+                this.blockedCraftingItems.add(material);
+            } catch (IllegalArgumentException ignored) {
+                Bukkit.getLogger().warning("[ServerService] Invalid material in blocked list: " + mat);
+            }
+        }
+    }
+
+    @Override
+    public void saveCraftingRecipes(Material material) {
+        FileConfiguration config = this.configService.getSettingsConfig();
+        File configFile = this.configService.getConfigFile("settings.yml");
+
+        config.set(this.BLOCKED_MATERIALS_SECTION, this.blockedCraftingItems.stream().map(Material::name).collect(Collectors.toList()));
+
+        this.configService.saveConfig(configFile, config);
+        this.updateCraftingRecipes();
+    }
+
+    @Override
+    public void addToBlockedCraftingList(Material material) {
+        if (!this.blockedCraftingItems.contains(material)) {
+            this.blockedCraftingItems.add(material);
+        }
+    }
+
+    @Override
+    public void removeFromBlockedCraftingList(Material material) {
+        this.blockedCraftingItems.remove(material);
+    }
+
+
+    @Override
+    public boolean isCraftable(Material material) {
+        ItemStack itemStack = new ItemStack(material);
+        return !Bukkit.getServer().getRecipesFor(itemStack).isEmpty();
+    }
+
+    @Override
+    public void updateCraftingRecipes() {
+        Bukkit.resetRecipes();
```

Comment:
this is wrong, you'll ruin custom crafts that are added by any other plugins and since this is toggle-able we want to allow custom crafts. iterator.remove(); also removes the recipe entirely from the server, why are we doing this? remove this functionality and rather have the event be cancelled. read up on PrepareItemCraftEvent, maybe smth like event.getInventory().setResult(null); if the resulting material is in the list